### DISCUSSION TITLE
Is it possible to run single precision MD with easy-installed deepmd and LAMMPS?

### DISCUSSION THREAD

**Discussion by cihe13375** [deepmodeling/deepmd-kit#1907] [Q&A] [answered]
Hi,

I'm trying to run a MD simulation on LAMMPS with deepmd models and I'm getting errors like this:
`2022-09-11 15:07:48.220590: F tensorflow/core/framework/tensor.cc:725] Check failed: dtype() == expected_dtype (1 vs. 2) double expected, got float
`

After searching in discussions it seems that this error raises because the model is in single precision (yes), but LAMMPS is built with double precision. I guess this can be solved by manually compile dpmd and LAMMPS (as illustrated in the doc), but I'm wondering if it's possible to run single precision simulations without manually compiling (i.e. by offline package or conda). It seems that the conda repo doesn't have `_low` variants anymore starting from version 2.0 (I searched by `conda search '*deepmd*[channel=https://conda.deepmodeling.com]'`). Is this intentional (like, maybe single precision is discouraged for general purposes)?

I have `export DP_INTERFACE_PREC=low` set before both training and MD.

Thanks!

**Comment by njzjz** [comment, marked as answer]
In the latest version, one can change the environment variable `LAMMPS_PLUGIN_PATH` from something like `/opt/deepmd-kit/lib/deepmd_lmp` to `/opt/deepmd-kit/lib/deepmd_lmp_low` to switch to single precision.

**Reply by cihe13375** [reply]
Thanks, it works.
